File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.java
class Solution {

    class Pair {
        int num;
        int freq;

        Pair(int num, int freq) {
            this.num = num;
            this.freq = freq;
        }
    }

    public int[] frequencySort(int[] nums) {

        int[] count = new int[201];
        for (var x : nums)
            count[x + 100]++;

        List<Pair> freqList = new ArrayList<>();

        for (int i = 0; i < count.length; i++) {
            if (count[i] > 0)
                freqList.add(new Pair(i - 100, count[i]));
        }

        freqList.sort((a, b) -> (a.freq == b.freq) ? b.num - a.num : a.freq - b.freq);

        int[] result = new int[nums.length];

        int index = 0;
        for (Pair p : freqList) {
            while (p.freq != 0) {
                result[index++] = p.num;
                p.freq--;
            }
        }

        return result;

    }
}

File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.py
from functools import cmp_to_key

class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        count = Counter(nums)
        def func(x1, x2):
             if(count[x1]==count[x2]):
                return x2 - x1 
             return count[x1] - count[x2]
        sort = sorted(nums, key = cmp_to_key(func))
        return sort